fido/fido.py

```diff
@@ -48,7 +41,7 @@ def _build_body_producer(body, headers):
     if not body:
         return None, headers
 
-    bodyProducer = FileBodyProducer(io.BytesIO(body))
+    bodyProducer = FileBodyProducer(io.BytesIO(encode_to_bytes(body)))
```

Comment:
this shouldn't change - body should always be bytes, if the user of this package is passing in text that should be a TypeError